controllers/places.js

```diff
@@ -0,0 +1,111 @@
+var _ = require('underscore'),
+    db = require('../db');
+
+var getPlace = function(id, callback) {
+  db.medic.get(id, function(err, doc) {
+    if (err) {
+      if (err.statusCode === 404) {
+        err.message  = 'Failed to find place.';
+      }
+      return callback(err);
+    }
+    callback(null, doc);
+  });
+};
+
+var isAPlace = function(place) {
+  return [
+    'district_hospital',
+    'health_center',
+    'clinic'
+  ].indexOf(place.type) !== -1;
+};
+
+/*
+ * Validate the basic data structure for a place.  Not checking against the
+ * database if parent exists because the entire child-parent structure might be
+ * created in one request. Just checking `type` field values and some required
+ * fields.
+ */
+var validatePlace = function(place, callback) {
+  if (!_.isObject(place)) {
+    return callback('Place must be an object.');
+  }
+  if (!isAPlace(place)) {
+    return callback('Wrong type, this is not a place.');
+  }
+  if (_.isUndefined(place.name)) {
+    return callback('Name property missing on place.');
+  }
+  if (!_.isString(place.name)) {
+    return callback('Property "name" must be a string.');
+  }
+  if (place.parent) {
+    // validate parents also
+    return validatePlace(place.parent, callback);
+  }
```

Comment:
Yep makes sense. 

We also had a `national_office` facility at one point, not sure what happened to that or if it's still in use.  It was useful theoretically to assign to the national admin role.  Probably for another day, or just another bandaid, but something I've been meaning to mention.